### PR TITLE
Roller bed holder update

### LINKED ISSUE #14997: Medborg Roller bed rack should change sprites when empty
(WEB REPORT BY: sonixapache REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)

d4360fb2c3ee6931875a3c97e51a8e77a06c3834 

>General description of the issue

you can't easily tell if you've got a bed on you or not
t. mouse (385)

### PR DESCRIPTION
**Pull request by jknpj**
deployed

borgbed
* Fixes holders not removing their references on Destroyed().
* Adds a new roller bed subype, the hover roller bed. Sprites by @Kokuten
* Updates the roller bed holder to start with a single hover roller bed. You can still hold a vanilla roller bed if you somehow lose your hover one.
* Updates the holder's icon and adds a DEPLOYED state to remember you to grab your bed when you're done. Sprites by @Kokuten

Fixes #14997

:cl:
 * rscadd: Adds a new roller bed subype, the hover roller bed. Sprites by @Kokuten
 * tweak: Updates mediborg's roller bed holder to start with a single hover roller bed. You can still hold a vanilla roller bed if you somehow lose your old one.
 * imageadd: Updates the holder's icon and adds a DEPLOYED state to remember you to grab your bed when you're done. Sprites by @Kokuten

### CODE REVIEW

**Comment by Zennyization** [comment]
Is there a difference in the new subtype of rollerbed? Or is it purely for aesthetic reasons?

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 1]
```diff
@@ -126,6 +126,9 @@
 	icon = 'icons/obj/rollerbed.dmi'
 	icon_state = "down"
 	anchored = 0
+	var/up_state ="up"
+	var/down_state = "down"
+	var/roller_type = "/obj/item/roller"
```
Comment (RemieRichards) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: don't put typepaths in strings.

**Comment by jknpj** [comment]
@Zennyization fluff

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 2]
```diff
@@ -187,16 +191,20 @@
 	if(istype(W,/obj/item/roller_holder))
 		var/obj/item/roller_holder/RH = W
 		if(!RH.held)
-			to_chat(user, "<span class='notice'>You collect the roller bed.</span>")
+			to_chat(user, "<span class='notice'>You collect the \the [src].</span>")
```
Comment (Intigracy) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: remove the `the` before the `\the`

Comment (jknpj) [reply code/game/objects/structures/stool_bed_chair_nest/bed.dm]: oh shit

**Comment by SonixApache** [comment]
add fixes #14997 to OP

**Pull request merged**